src/Framework/Assert.php

```diff
@@ -2524,6 +2526,34 @@ public static function assertJsonFileNotEqualsJsonFile(string $expectedFile, str
         static::assertThat($actualJson, new LogicalNot($constraintExpected), $message);
     }
 
+    /**
+     * Asserts that given value is resource of specified type
+     *
+     * @param string $expectedType
+     * @param resource $actual
+     * @param string $message
+     *
+     * @throws \Exception
+     */
+    public static function assertResourceOfType(string $expectedType, $actual, string $message = '')
+    {
+        static::assertThat($actual, new IsResourceOfType($expectedType), $message);
+    }
+
+    /**
+     * Asserts that given value is resource not of specified type
+     *
+     * @param string $expectedType
+     * @param resource $actual
+     * @param string $message
+     *
+     * @throws \Exception
+     */
+    public static function assertResourceNotOfType(string $expectedType, $actual, string $message = '')
+    {
+        static::assertThat($actual, new IsResourceNotOfType($expectedType), $message);
```

Comment:
this constraint is highly duplicating positive one.
take a note that there is no negative constraint so far.
drop that extra constraint and instead of it, use `LogicalNot`

Reply:
I know, but `IsResourceOfType` uses `IsType` internally.
Because of that I cannot use `LogicalNot` here because in both cases when `$actual` is not a resource match should fail.

Is it ok or should I try with different design?

Reply:
I do not think that it is a good idea to use `IsType` in `IsResourceOfType`.

Reply:
I can stop using `IsType` inside `IsResourceOfType`, but `IsResourceNotOfType` is not an exact opposite of `IsResourceOfType`. `IsNotResourceOfType` would be.

`IsResourceOfType` means that value is a resource of specific type.
`IsResourceNotOfType` means that value is a resource not of specific type.

Both still must be a resource.

`IsNotResourceOfType` sounds like an exact opposite, so it's either not a resource or a resource of different type.

That's why I introduced `IsResourceNotOfType` as a separate constraint.

Should we have 2 separate constraints:
- `IsResourceOfType`
- `IsResourceNotOfType`

and 3 assertions:
- `assertResourceOfType`
- `assertResourceNotOfType`
- `assertNotResourceOfType`

Or having only `IsResourceOfType` with 2 assertions: `assertResourceOfType` and `assertNotResourceOfType` is sufficent? 

Imo `IsResourceNotOfType` is useful.